### PR TITLE
Read multiple values at once

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: storage/mysql/map_storage.go [hunk 1]
```diff
@@ -115,31 +115,47 @@ func (m *mapTX) Set(keyHash trillian.Hash, value trillian.MapLeaf) error {
...
-		&mapKeyHash, &mapRevision, &flatData)
+	args := make([]interface{}, 0, len(keyHashes)+2)
+	for _, k := range keyHashes {
+		args = append(args, []byte(k[:]))
+	}
+	args = append(args, m.ms.mapID.TreeID)
+	args = append(args, revision)
 
+	rows, err := stx.Query(args...)
 	// It's possible there is no value for this value yet
```
Comment (Martin2112) [inline storage/mysql/map_storage.go]: Can you tidy this comment up while you're in there. "value for a value"?

File: server/vmap/trillian_map_server.go [hunk 2]
```diff
@@ -102,19 +102,30 @@ func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapL
...
 
+	keyHashes := make([]trillian.Hash, 0, len(req.Key))
+	hashToKey := make(map[string][]byte)
 	for _, key := range req.Key {
 		kHash := kh.HashKey(key)
+		keyHashes = append(keyHashes, kHash)
+		hashToKey[string(kHash)] = key
+	}
+
+	leaves, err := tx.Get(req.Revision, keyHashes)
```
Comment (AlCutter) [inline server/vmap/trillian_map_server.go]: No, it'll return what it had - the caller must use the `KeyHash` entry in the leaf struct to determine the association between key/value.

File: storage/mysql/storage_test.go [hunk 3]
```diff
@@ -1012,9 +1012,12 @@ func TestMapGetUnknownKey(t *testing.T) {
 	{
 		tx := beginMapTx(s, t)
 
-		readValue, err := tx.Get(1, []byte("This doesn't exist."))
-		if got, want := err, storage.ErrNoSuchKey; got != want {
-			t.Fatalf("Read %v with error %v, but expected error %v", readValue, got, want)
+		readValues, err := tx.Get(1, []trillian.Hash{[]byte("This doesn't exist.")})
+		if err != nil {
+			t.Fatalf("Read returnd error %v", err)
```
Comment (Martin2112) [inline storage/mysql/storage_test.go]: typo I missed first time.

File: storage/mysql/map_storage.go (hunk 1 above)
Comment (AlCutter) [inline storage/mysql/map_storage.go]: done

File: storage/mysql/storage_test.go (hunk 3 above)
Comment (AlCutter) [inline storage/mysql/storage_test.go]: fixed

**Pull request merged**